app/models/pipeline_run.rb

```diff
@@ -113,6 +115,18 @@ def active_stage
     nil
   end
 
+  def retry
+    return unless failed? # only resuming from a failed job
+    self.finalized = 0
+    prs = active_stage
```

Comment:
I suggest we leave the state transitions & kicking off the job to the pipeline monitor.

```
def retry
    return unless failed?
    prs = active_stage
    prs.job_status = nil
    prs.job_command = nil
    prs.save
    self.finalized = 0
    save
```